zrcs/zrc-2.md

```diff
@@ -0,0 +1,412 @@
+```
+
+|        | Name         | Type      | Description                                         |
+| ------ | ------------ | --------- | --------------------------------------------------- |
+| @param | `operator`   | `ByStr20` | Address of a default operator.                      |
+| @param | `recipient`  | `ByStr20` | Address of the recipient whose balance is increased.|
+| @param | `amount`     | `Uint128` | Number of tokens to be minted.                      |
+
+|           | Name                       | Description                             | Event Parameters                                                                                                                                                                                                               |
+| --------- | -------------------------- | --------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| eventName | `OperatorMintAllSuccess` | Minting is successful.     | `recipient`: `ByStr20`, and `amount`: `Uint128`. |
+| eventName | `Error`                    | Minting is not successful. | - emit `CodeNotAuthorised` if the transition is not called by an approved operator.                                                                                                      |
+
+
+#### 9. AuthorizeOperator
+
+```ocaml
+(* @dev: Make an address an operator of the caller.                           *)
+(* @param operator: Address to be set as operator. Cannot be calling address. *)
+transition AuthorizeOperator(operator: ByStr20)
+```
+
+|        | Name       | Type      | Description                    |
+| ------ | ---------- | --------- | ------------------------------ |
+| @param | `operator` | `ByStr20` | Address to be set as operator. Cannot be calling address. |
+
+|           | Name                       | Description                             | Event Parameters                                                                                                                                                                                                               |
+| --------- | -------------------------- | --------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| eventName | `AuthorizeOperatorSuccess` | Authorizing is successful.     | `operator`: `ByStr20`. |
+| eventName | `Error`                    | Authorizing is not successful. | - emit `CodeNotAuthorised` if the transition is called by a user who is not the token holder.                                                                                                      |
```

Comment:
I'm not sure I get the `Error` part. Anybody should be able to authorize anybody else as an operator right? Why should the error be "CodeNotAuthorized" ? There is no "token" to be considered here for someone to be "token holder".

Reply:
@vaivaswatha The sender cannot make himself/herself the operator.